data-otservbr-global/scripts/creaturescripts/others/login.lua

```diff
@@ -258,6 +258,13 @@ function playerLogin.onLogin(player)
 		onExerciseTraining[player:getId()] = nil
 		player:setTraining(false)
 	end
+
+	-- check user won exercise weapon and send message
+	if not table.contains({ TOWNS_LIST.DAWNPORT, TOWNS_LIST.DAWNPORT_TUTORIAL }, player:getTown():getId()) then
+		if player:getStorageValue(tonumber(Storage.PlayerWeaponReward)) ~= 1 then
+			player:sendTextMessage(MESSAGE_LOGIN, "You can win an exercise weapon using command !reward")
+		end
+	end
```

Comment:
This should be an optional feature. Admins should be able to turn on the talk action and this message, default should be off. 

You can put this in it's own CreatureEvent in a new file where you keep both the config and this event. That way it stays separate from basic login logic.